wp-modules/pattern-preview-renderer/pattern-preview-renderer.php

```diff
@@ -0,0 +1,133 @@
+
+	wp_head();
+
+	?>
+	<body style="overflow: hidden;" <?php body_class(); ?>>
+	<?php
+
+	do_action( 'patternmanager_before_pattern_preview' );
+
+	the_content();
+
+	do_action( 'patternmanager_after_pattern_preview' );
+
+	wp_footer();
+
+	?>
+	</body>
+	<?php
+
+	exit;
+}
+add_action( 'wp', __NAMESPACE__ . '\display_block_pattern_preview', PHP_INT_MAX );
+
+/**
+ * Create a fake post object for this pattern preview, and set it as the global $post for WordPress.
+ *
+ * @param string $pattern_content The block pattern raw html.
+ */
+function mock_pattern_preview_post_object( $pattern_content ) {
+	global $wp, $wp_query;
+
+	$post_id              = -998; // negative ID, to avoid clash with a valid post.
+	$post                 = new \stdClass();
+	$post->ID             = $post_id;
+	$post->post_author    = 1;
+	$post->post_date      = current_time( 'mysql' );
+	$post->post_date_gmt  = current_time( 'mysql', 1 );
+	$post->post_title     = 'Pattern Preview';
+	$post->post_content   = $pattern_content;
+	$post->post_status    = 'publish';
+	$post->comment_status = 'closed';
+	$post->ping_status    = 'closed';
+	$post->post_name      = 'pattern-preview-' . wp_rand( 1, 99999 ); // append random number to avoid clash.
+	$post->post_type      = 'page';
+	$post->filter         = 'raw';
+
+	// Convert to WP_Post object.
+	$wp_post = new \WP_Post( $post );
+
+	// Add the fake post to the cache.
+	wp_cache_add( $post_id, $wp_post, 'posts' );
+
+	// Update the main query.
+	$wp_query->post                 = $wp_post;
+	$wp_query->posts                = array( $wp_post );
+	$wp_query->queried_object       = $wp_post;
+	$wp_query->queried_object_id    = $post_id;
+	$wp_query->found_posts          = 1;
+	$wp_query->post_count           = 1;
+	$wp_query->max_num_pages        = 1;
+	$wp_query->is_page              = true;
+	$wp_query->is_singular          = true;
+	$wp_query->is_single            = false;
+	$wp_query->is_attachment        = false;
+	$wp_query->is_archive           = false;
+	$wp_query->is_category          = false;
+	$wp_query->is_tag               = false;
+	$wp_query->is_tax               = false;
+	$wp_query->is_author            = false;
+	$wp_query->is_date              = false;
+	$wp_query->is_year              = false;
+	$wp_query->is_month             = false;
+	$wp_query->is_day               = false;
+	$wp_query->is_time              = false;
+	$wp_query->is_search            = false;
+	$wp_query->is_feed              = false;
+	$wp_query->is_comment_feed      = false;
+	$wp_query->is_trackback         = false;
+	$wp_query->is_home              = false;
+	$wp_query->is_embed             = false;
+	$wp_query->is_404               = false;
+	$wp_query->is_paged             = false;
+	$wp_query->is_admin             = false;
+	$wp_query->is_preview           = false;
+	$wp_query->is_robots            = false;
+	$wp_query->is_posts_page        = false;
+	$wp_query->is_post_type_archive = false;
```

Comment:
Sorry to comment after merge, but are these `= false;` statements needed?

Many are `false` [by default](https://github.com/WordPress/wordpress-develop/blob/a5f3087f6b5d9c52dbe67ed247165dc32427c57d/src/wp-includes/class-wp-query.php#L194-L454).